addons/hr_timesheet/project_timesheet.py

```diff
@@ -59,3 +60,9 @@ def _hours_get(self):
     subtask_count = fields.Integer(compute='_get_subtask_count', type='integer', string="Sub-task count")
 
     _constraints = [(models.BaseModel._check_recursion, 'Circular references are not permitted between tasks and sub-tasks', ['parent_id'])]
+
+    @api.constrains('parent_id')
+    def _check_subtask_project(self):
+        for task in self:
+            if task.parent_id and task.parent_id.project_id and task.project_id and task.project_id != task.parent_id.project_id.subtask_project_id:
+                raise UserError(_("Either you forgot to define a sub-task project on a parent task's project or make sure the project in sub-task project and the project of your current task are same"))
```

Comment:
Please make sure your task is not an issue for your project; otherwise please open a ticket in our issue tracker so that someone assigns the task to a project manager.